scripts/build-aci

```diff
@@ -26,79 +26,38 @@ if [ -z "$VERSION" ] ; then
     exit 1
 fi
 
-mkdir -p $IMAGEDIR/rootfs
-if [ ! -d $IMAGEDIR/rootfs -o ! -x $IMAGEDIR/rootfs ]; then
-    echo "$IMAGEDIR/rootfs is not a writeable directory"
-    exit 1
-fi
-
-if [ -n "$(ls -A $IMAGEDIR/rootfs)" ]; then
-    echo "$IMAGEDIR/rootfs is not empty"
-    exit 1
-fi
+acbuild --debug begin
 
-cp $BINARYDIR/etcd $BINARYDIR/etcdctl $IMAGEDIR/rootfs/
-cp README.md            $IMAGEDIR/rootfs/
-cp etcdctl/README.md    $IMAGEDIR/rootfs/README-etcdctl.md
-cp -r Documentation     $IMAGEDIR/rootfs/
-
-cat <<DF > $IMAGEDIR/manifest
-{
-    "acVersion": "0.5.1",
-    "acKind": "ImageManifest",
-    "name": "coreos.com/etcd",
-    "labels": [
-        {"name": "os", "value": "linux"},
-        {"name": "arch", "value": "amd64"},
-        {"name": "version", "value": "${VERSION}"}
-    ],
-    "app": {
-        "environment": [
-             {
-                    "name": "ETCD_DATA_DIR",
-                    "value": "/data-dir"
-             }
-        ],
-        "exec": [
-             "/etcd"
-        ],
-        "user": "0",
-        "group": "0",
-        "mountPoints": [
-             {
-                    "name": "data-dir",
-                    "path": "/data-dir"
-             }
-        ],
-        "ports": [
-            {
-                "name": "legacy-client",
-                "port": 4001,
-                "protocol": "tcp"
-            },
-            {
-                "name": "client",
-                "port": 2379,
-                "protocol": "tcp"
-            },
-            {
-                "name": "legacy-peer",
-                "port": 7001,
-                "protocol": "tcp"
-            },
-            {
-                "name": "peer",
-                "port": 2380,
-                "protocol": "tcp"
-            }
-        ]
-    }
+acbuildEnd() {
+    export EXIT=$?
+    acbuild --debug end && exit $EXIT 
 }
-DF
+trap acbuildEnd EXIT
+
+acbuild --debug set-name coreos.com/etcd
+
+acbuild --debug copy --to-dir $BINARYDIR/etcd $BINARYDIR/etcdctl /
+acbuild --debug copy README.md /README.md
+acbuild --debug copy etcdctl/README.md /README-etcdctl.md
+acbuild --debug copy --to-dir Documentation /
+
+acbuild --debug environment add ETCD_DATA_DIR /data-dir
+
+acbuild --debug label add version "$VERSION"
+
+acbuild --debug set-exec -- /etcd
+
+acbuild --debug mount add data-dir /data-dir
+
+acbuild --debug port add legacy-client tcp 4001
+acbuild --debug port add legacy-peer tcp 7001
+acbuild --debug port add client tcp 2379
+acbuild --debug port add peer tcp 2380
 
-mkdir -p $IMAGEDIR/rootfs/etc/
-cat <<DF > $IMAGEDIR/rootfs/etc/hosts
+cat <<DF > hosts
 127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
 DF
+acbuild --debug copy hosts /etc/hosts
+rm hosts
```

Comment:
Moved the `cat` to right after `acbuild begin`, and moved the `rm` to `acbuildEnd`, so it'll get called when the script exits.